### PR TITLE
Fix Issue #28

### LINKED ISSUE #28: Bug: Deletion of SIB results in crash
## Description
After successfully applying the SecurityIntent (SI) and SecurityIntentBinding (SIB), if we try to delete the SIB then nimbus crashes. However, SIB gets deleted from the k8s API server. 

## Steps to reproduce
1. Run nimbus `make run`
2. Start HTTP server `go run pkg/nimbus-kubearmor/receiver/server/server.go`
3. Apply SI and SIB 
4. Delete SIB
And then the nimbus crashes.

## Expected behaviour
nimbus should not crash on the deletion of SIB.

## Additional context
Screenshot 2024-01-05 at 14 26 11

### PR DESCRIPTION
**Pull request by seungsoo-lee**
## Description

Fixes #28 

**Does this PR introduce a breaking change?**

## Checklist

- [ ] PR title follows the `<type>: <description>` convention
- [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
- [ ] I have updated the [documentation](../docs) accordingly
- [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
- [ ] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pkg/cleanup/cleanup.sh
Comment (anurag-rajawat) [inline pkg/cleanup/cleanup.sh:1]: What is the specific purpose of this script in explicitly removing the CRs? Under what circumstances should a user execute it?

**Comment by anurag-rajawat** [comment]
Please squash commits.

**Review by rajaSahil — APPROVED** [review]
@seungsoo-lee PR looks good, please squash the commits.

File: api/v1/securityintentbinding_types.go
Comment (anurag-rajawat) [inline api/v1/securityintentbinding_types.go:1]: We don't need these sophisticated parameters in SecurityIntent; if there is a need for parameters, we'll specify those generically.

**Comment by anurag-rajawat** [comment]
@seungsoo-lee, we can close this PR since the changes have been split into two separate PRs for better organization and review:
https://github.com/5GSEC/nimbus/pull/30 resolves the issues addressed in #28.
https://github.com/5GSEC/nimbus/pull/31  introduces the new features."

**Pull request closed without merging**